docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -27,27 +27,35 @@ These instructions show how to enable and resolve static analysis warnings to pr
+You can set `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` (in .NET 6+) in a `<PropertyGroup>` tag in your library project file. This will not have any effect on the output (it will not have metadata indicating it is trimmable), but it will enable trim analysis during build via the Roslyn analyzer.
 
-### Show all warnings
+#### Set `IsTrimmable`
 
-To show all analysis warnings for your library, including warnings about dependencies, create a separate app project like the following that references your library, and publish it with `PublishTrimmed`.
+Set `<IsTrimmable>true</IsTrimmable>` (in .NET 6+) in a `<PropertyGroup>` tag your library project file. This will mark your assembly as "trimmable". Being "trimmable" means when your library is used in a trimmed application, the assembly can have its unused members trimmed in the final output. Otherwise, the entire assembly will be kept.
 
-The extra step of creating an app project just to get complete warnings for a library is necessary because the implementations of dependencies are not generally available during `dotnet build`, and reference assemblies don't contain enough information to determine whether they are compatible with trimming. Publishing a self-contained app ensures that the library is analyzed in a context where its dependencies are available, so that you are alerted if your library uses any code from dependencies that could break a trimmed app.
+### Show all warnings with sample application
+
+To show all analysis warnings for your library, including warnings about dependencies, you need the trimmer to analyze the implementation of your library and the implementations of dependencies your library uses. When building and publishing a library, the implementations of the dependencies are not available, and the reference assemblies that are available do not have enough information for the trimmer to determine if they are compatible with trimming. Because of this, you'll need to create and publish a self-contained sample application which produces an executable that includes your library and the dependencies it relies on. This executable includes all the information the trimmer requires to warn you about all trim incompatibilities in your library code, as well as the code that your library references from its dependencies.
+
+To create your sample app, first create a separate application project with `dotnet new` and modify the project file to look like the following. You'll need to include a reference to your library, as well as add `<TrimmerRootAssembly Include="YourLibraryName" />` in an `<ItemGroup>` tag, and `<TrimmerDefaultAction>link</TrimmerDefaultAction>` in a `<PropertyGroup>` tag. No changes to the source code are necessary. Once your project file is updated, run `dotnet publish` with the [runtime identifier (RID)](../../rid-catalog.md) you want to target, and `net6.0` as the target framework.
```

Comment:
Maybe mention `PublishTrimmed` here too. I think this section would be more readable as a bullet list. I'd like this section to start with a pretty concise explanation+example of how to enable this for people who just want to reference the instructions. Then I'd put more detailed explanation after the example. How about this?:

To create your sample app, first ... add the following to your project file:
- Set `<PublishTrimmed>true</PublishTrimmed>`
- Include a reference to your library
- Add `<TrimmerRootAssembly`... to ensure that every part of the library is analyzed.
- Add `<TrimmerDefaultAction` ... to prevent analysis of unused parts of assemblies which your library depends on.

... (project file and publish command) ...

Then later include more details about what `TrimmerRootAssembly` and `TrimmerDefaultAction` do, and remove the bullets there.

Reply:
I agree, that's much cleaner.